securedrop_client/logic.py

```diff
@@ -251,3 +253,36 @@ def logout(self):
         """
         self.api = None
         self.gui.logout()
+
+    def on_file_click(self, message):
+        """
+        Download the file associated with the associated message (which may
+        be a Submission or Reply).
+        """
+        if isinstance(message, sdclientapi.Submission):
+            # Handle sources.
+            func = self.api.download_submission
+        else:
+            # Handle journalist's replies.
+            func = self.api.download_reply
+        self.call_api(func, self.on_file_download,
+                      self.on_download_timeout, message, self.data_dir)
+
+    def on_file_download(self):
+        """
+        Called when a file has downloaded. Cause a refresh to the conversation
+        view to display the contents of the new file.
```

Comment:
though we will just indicate that the file is downloaded instead of opening (the actual opening logic will be dealt with in #20), i.e. "download" before DL and "view" after DL (following the language in the relevant snippets from wireframe):

<img width="446" alt="screen shot 2018-10-24 at 6 30 06 pm" src="https://user-images.githubusercontent.com/7832803/47470505-e58e5780-d7ba-11e8-8c40-29a55d950156.png">

<img width="466" alt="screen shot 2018-10-24 at 6 30 16 pm" src="https://user-images.githubusercontent.com/7832803/47470506-e58e5780-d7ba-11e8-8c12-79adfcacd738.png">

Reply:
Right... so I just want to be *absolutely certain* I grok this:

* There are two sorts of files that can be downloaded and then decrypted: 1) Plaintext files containing a message (from now on "messages"). 2) Attachments files (such a images or PDF docs -- from now on "attachments").
* If a file isn't downloaded we just show that it's ready to be downloaded and encrypted. I'm assuming there's a way to distinguish between messages and attachments due to filename naming conventions, right?
* Once a file (no matter if it is a message or attachment) is downloaded, then this function we will display the contents of the downloaded and decrypted messages as the speech bubbles and the downloaded attachment as a file widget in a viewable/exportable state.
* All pending-download/encryption messages/attachments will just look like the "download" image in your comment.

Sorry for the painful description of detail, but it's important to get this written out somewhere. ;-)